### ISSUE TITLE
Unexpected Runtime Exception when skipping video

### ISSUE THREAD

**Issue by bobMuk** [bennettpeter/android-MythTV-Leanfront#8]
mythtvbackend 31 running the current latest version of leanfront (leanfront-v0-218-gd488336) on a Fire TV 4K

Certain recordings - it appears to be anything in HD from UK Freeview - give a runtime exception when I attempt to skip the video (in either direction).  The exception happens on every skip, though doesn't cause playback to fail and the skip does occur, but there's a lag of upto 10 seconds from pressing the button, through the exception, to the video continuing to play.  I've also noticed that starting up playback takes aprox 10 seconds whereas other recordings are far quicker to start playing (though I don't see the onscreen exception message at the start of playback).

--------- beginning of system
--------- beginning of main
10-29 18:26:30.997  3716  4895 I lfe     : AsyncBackendCall Time difference 3 milliseconds
10-29 18:26:38.034  3716  3716 I lfe     : MainActivity memoryClass:192 lMemoryClass:384
10-29 18:26:38.446  3716  3780 I lfe     : AsyncBackendCall Time difference -446 milliseconds
10-29 18:26:51.285  3716  4895 I lfe     : AsyncBackendCall Time difference -285 milliseconds
10-29 18:26:56.966  3716  3716 I lfe     : PlaybackFragment Initializing Player for Question Time http://192.168.1.6:6544/Content/GetFile?StorageGroup=Default&FileName=/20001_20201022214300.ts
10-29 18:26:57.080  3716  3716 I lfe     : PlaybackFragment Playing offset mSec:0
10-29 18:26:58.095  3716  3716 I lfe     : PlaybackFragment File Length changed from -1 to 2030519192
10-29 18:27:09.214  3716  3716 E lfe     : PlaybackFragment Player Error Question Time http://192.168.1.6:6544/Content/GetFile?StorageGroup=Default&FileName=/20001_20201022214300.ts
10-29 18:27:09.214  3716  3716 E lfe     : c.b.a.a.k0: Unexpected runtime error
10-29 18:27:09.214  3716  3716 E lfe     :      at c.b.a.a.m0.handleMessage()
10-29 18:27:09.214  3716  3716 E lfe     :      at android.os.Handler.dispatchMessage(Handler.java:98)
10-29 18:27:09.214  3716  3716 E lfe     :      at android.os.Looper.loop(Looper.java:154)
10-29 18:27:09.214  3716  3716 E lfe     :      at android.os.HandlerThread.run(HandlerThread.java:61)
10-29 18:27:09.214  3716  3716 E lfe     : Caused by: java.lang.IllegalStateException: Playback stuck buffering and not loading
10-29 18:27:09.214  3716  3716 E lfe     :      at c.b.a.a.m0.l()
10-29 18:27:09.214  3716  3716 E lfe     :      ... 4 more
10-29 18:27:09.214  3716  3716 E lfe     : PlaybackFragment Player Error Unexpected Runtime Exception
10-29 18:27:09.214  3716  3716 E lfe     : Unexpected runtime error
10-29 18:27:09.214  3716  3716 E lfe     : Playback stuck buffering and not loading
10-29 18:27:09.251  3716  3716 I lfe     : PlaybackFragment Playing offset mSec:0
10-29 18:27:10.274  3716  3716 I lfe     : PlaybackFragment File Length changed from -1 to 2030519192
10-29 18:27:22.175  3716  3716 E lfe     : PlaybackFragment Player Error Question Time http://192.168.1.6:6544/Content/GetFile?StorageGroup=Default&FileName=/20001_20201022214300.ts
10-29 18:27:22.175  3716  3716 E lfe     : c.b.a.a.k0: Unexpected runtime error
10-29 18:27:22.175  3716  3716 E lfe     :      at c.b.a.a.m0.handleMessage()
10-29 18:27:22.175  3716  3716 E lfe     :      at android.os.Handler.dispatchMessage(Handler.java:98)
10-29 18:27:22.175  3716  3716 E lfe     :      at android.os.Looper.loop(Looper.java:154)
10-29 18:27:22.175  3716  3716 E lfe     :      at android.os.HandlerThread.run(HandlerThread.java:61)
10-29 18:27:22.175  3716  3716 E lfe     : Caused by: java.lang.IllegalStateException: Playback stuck buffering and not loading
10-29 18:27:22.175  3716  3716 E lfe     :      at c.b.a.a.m0.l()
10-29 18:27:22.175  3716  3716 E lfe     :      ... 4 more
10-29 18:27:22.175  3716  3716 E lfe     : PlaybackFragment Player Error Unexpected Runtime Exception
10-29 18:27:22.175  3716  3716 E lfe     : Unexpected runtime error
10-29 18:27:22.175  3716  3716 E lfe     : Playback stuck buffering and not loading
10-29 18:27:22.180  3716  3716 I lfe     : PlaybackFragment Playing offset mSec:0
10-29 18:27:23.217  3716  3716 I lfe     : PlaybackFragment File Length changed from -1 to 2030519192

I guess that "Playback stuck buffering and not loading" might imply the issue is with the backend, but the video plays fine in the native mythfrontend.

**Comment by bennettpeter** [comment]
Freeview UK has some problems. Before v0-207-g11262a3 they would not play at all. I have one recording from freeview UK that I obtained from a UK user, and I am not able to recreate your error. Is it possible that it is only happening with LiveTV or with a recording in progress? I am unable to test those since I am in the USA.

**Comment by bobMuk** [comment]
It's neither live nor is there a recording in progress when I'm playing. I'll do a bit more testing but it does seem to be HD recordings only (and possibly only from BBC transports).  If you're interested in taking a look I'm more than happy to provide an example recording.
(The reason this one's a biggie for me is that most of my new recordings are HD/BBC)

**Comment by bobMuk** [comment]
Confirmed it's any HD recordings - VLC says:

Codec: H264 - MPEG-4 AVC (part 10) (h264)
Type: Video
Video resolution: 1920x1080
Buffer dimensions: 1920x1088
Frame rate: 25

**Comment by bennettpeter** [comment]
Please test on the new version from today (leanfront-v0-222-g0428d75.apk). You will need to enable Settings Playback -> Show Playback Error Toast to see the Exception error message. If the error still happens please supply a sample video, a couple of minutes that will allow testing of skips back and forward. You can extract part of a recording with dd, for example

`dd bs=1MB count=200 if=3706_20180529000000.ts_org of=extract.ts`

add skip=nn if needed to skip nn MB before the start of the extract.

**Comment by bobMuk** [comment]
Thanks, I'll give it a go as soon as I can - at the moment I can only find leanfront-v0-220-g9dcb478-dirty.apk at https://dl.bintray.com/bennettpeter/generic/mythtv_leanfront/android

**Comment by bennettpeter** [comment]
Sorry, I released the wrong file. I have replaced it now.

**Comment by bobMuk** [comment]
Thanks Peter.  I still get the error with 222 - the log is:

--------- beginning of system
--------- beginning of main
11-05 18:08:41.180  3074  3074 E lfe     : PlaybackFragment Player Error University Challenge http://192.168.1.6:6544/Content/GetFile?StorageGroup=Default&FileName=/20102_20201102202800.ts
11-05 18:08:41.180  3074  3074 E lfe     : c.b.a.a.k0: Unexpected runtime error
11-05 18:08:41.180  3074  3074 E lfe     :      at c.b.a.a.m0.handleMessage()
11-05 18:08:41.180  3074  3074 E lfe     :      at android.os.Handler.dispatchMessage(Handler.java:98)
11-05 18:08:41.180  3074  3074 E lfe     :      at android.os.Looper.loop(Looper.java:154)
11-05 18:08:41.180  3074  3074 E lfe     :      at android.os.HandlerThread.run(HandlerThread.java:61)
11-05 18:08:41.180  3074  3074 E lfe     : Caused by: java.lang.IllegalStateException: Playback stuck buffering and not loading
11-05 18:08:41.180  3074  3074 E lfe     :      at c.b.a.a.m0.l()
11-05 18:08:41.180  3074  3074 E lfe     :      ... 4 more
11-05 18:08:41.180  3074  3074 E lfe     : PlaybackFragment Player Error Unexpected Runtime Exception
11-05 18:08:41.180  3074  3074 E lfe     : Unexpected runtime error
11-05 18:08:41.180  3074  3074 E lfe     : Playback stuck buffering and not loading
11-05 18:08:41.185  3074  3074 I lfe     : PlaybackFragment Playing offset mSec:0
11-05 18:08:42.219  3074  3074 I lfe     : PlaybackFragment File Length changed from -1 to 1016852896

I've uploaded a sample video that's aprox 2.5 minutes long to https://filebin.net/e8a9f6t6mb7684q9
Thanks :)

**Comment by bennettpeter** [comment]
There is only one exception reported, at the beginning of playback. This is the known error caused by the second audio track with no audio data. The application catches the error and fixes it by nullifying the second audio track. I suppose in that case the exception error message could be suppressed since the exception is handled. Skipping back and forward does not cause any exceptions, although it is slow. I don't know why skipping is so slow on this.

**Comment by bobMuk** [comment]
That's odd, the exception that occurred in the section of log output from today occurred during a skip, ie. not at the start of playback - I started playback, did a few skips, connected via adb and did another skip which then produced the log output. Is it possible that you're seeing a different behaviour because it's only a section of the video and not the whole thing?

**Comment by bennettpeter** [comment]
Is this still causing a problem? 

Do you get unexpected runtime exceptions that cause a failure, or does the system recover from them? 

If it recovers from the exception and continues playback uninterrupted, then the solution is to avoid the messages by disabling "show playback error toast" in the settings.

**Comment by bobMuk** [comment]
Hi - yes, still have issues.
I've just installed the latest version - leanfront-v0-236-gcac3df8.apk - and tried again.

I'm getting runtime exceptions and playback failed.  Clicking continue just gives another failure.
This appears to be worse than what I was seeing with 222, where I was getting the exceptions but playback would then continue.

Here's the log:


--------- beginning of system
--------- beginning of main
12-17 08:53:12.449  5827  5827 I lfe     : MainActivity memoryClass:192 lMemoryClass:384
12-17 08:53:13.516  5827  5862 I lfe     : AsyncBackendCall Time difference -516 milliseconds
12-17 08:53:38.633  5827  5924 I lfe     : AsyncBackendCall Time difference -633 milliseconds
12-17 08:53:54.719  5827  5936 I lfe     : AsyncBackendCall Time difference -719 milliseconds
12-17 08:54:00.249  5827  5827 I lfe     : PlaybackFragment Initializing Player for Question Time http://192.168.1.6:6544/Content/GetFile?StorageGroup=Default&FileName=/20101_20201210224300.ts
12-17 08:54:00.373  5827  5827 I lfe     : PlaybackFragment Playing offset mSec:0
12-17 08:54:01.393  5827  5827 I lfe     : PlaybackFragment File Length changed from -1 to 2178983920
12-17 08:54:01.887  5827  5827 E lfe     : PlaybackFragment Player Error Question Time http://192.168.1.6:6544/Content/GetFile?StorageGroup=Default&FileName=/20101_20201210224300.ts
12-17 08:54:01.887  5827  5827 E lfe     : c.b.a.a.k0: Unexpected runtime error
12-17 08:54:01.887  5827  5827 E lfe     :      at c.b.a.a.m0.handleMessage()
12-17 08:54:01.887  5827  5827 E lfe     :      at android.os.Handler.dispatchMessage(Handler.java:98)
12-17 08:54:01.887  5827  5827 E lfe     :      at android.os.Looper.loop(Looper.java:154)
12-17 08:54:01.887  5827  5827 E lfe     :      at android.os.HandlerThread.run(HandlerThread.java:61)
12-17 08:54:01.887  5827  5827 E lfe     : Caused by: java.lang.IllegalStateException: Playback stuck buffering and not loading
12-17 08:54:01.887  5827  5827 E lfe     :      at c.b.a.a.m0.l()
12-17 08:54:01.887  5827  5827 E lfe     :      ... 4 more
12-17 08:54:01.888  5827  5827 E lfe     : PlaybackFragment Player Error Unexpected Runtime Exception
12-17 08:54:01.888  5827  5827 E lfe     : Unexpected runtime error
12-17 08:54:01.888  5827  5827 E lfe     : Playback stuck buffering and not loading
12-17 08:54:01.904  5827  5827 I lfe     : PlaybackFragment Playing offset mSec:0
12-17 08:54:02.919  5827  5827 I lfe     : PlaybackFragment File Length changed from -1 to 2178983920

*** Skipped forward the get past commercials - saw exception and several seconds lag but playback continued

12-17 08:54:31.604  5827  5827 E lfe     : PlaybackFragment Player Error Question Time http://192.168.1.6:6544/Content/GetFile?StorageGroup=Default&FileName=/20101_20201210224300.ts
12-17 08:54:31.604  5827  5827 E lfe     : c.b.a.a.k0: Unexpected runtime error
12-17 08:54:31.604  5827  5827 E lfe     :      at c.b.a.a.m0.handleMessage()
12-17 08:54:31.604  5827  5827 E lfe     :      at android.os.Handler.dispatchMessage(Handler.java:98)
12-17 08:54:31.604  5827  5827 E lfe     :      at android.os.Looper.loop(Looper.java:154)
12-17 08:54:31.604  5827  5827 E lfe     :      at android.os.HandlerThread.run(HandlerThread.java:61)
12-17 08:54:31.604  5827  5827 E lfe     : Caused by: java.lang.IllegalStateException: Playback stuck buffering and not loading
12-17 08:54:31.604  5827  5827 E lfe     :      at c.b.a.a.m0.l()
12-17 08:54:31.604  5827  5827 E lfe     :      ... 4 more
12-17 08:54:31.606  5827  5827 E lfe     : PlaybackFragment Player Error Unexpected Runtime Exception
12-17 08:54:31.606  5827  5827 E lfe     : Unexpected runtime error
12-17 08:54:31.606  5827  5827 E lfe     : Playback stuck buffering and not loading
12-17 08:54:31.616  5827  5827 I lfe     : PlaybackFragment Playing offset mSec:0
12-17 08:54:32.630  5827  5827 I lfe     : PlaybackFragment File Length changed from -1 to 2178983920

*** skipped forwards again

12-17 08:54:39.941  5827  5827 E lfe     : PlaybackFragment Player Error Question Time http://192.168.1.6:6544/Content/GetFile?StorageGroup=Default&FileName=/20101_20201210224300.ts
12-17 08:54:39.941  5827  5827 E lfe     : c.b.a.a.k0: Unexpected runtime error
12-17 08:54:39.941  5827  5827 E lfe     :      at c.b.a.a.m0.handleMessage()
12-17 08:54:39.941  5827  5827 E lfe     :      at android.os.Handler.dispatchMessage(Handler.java:98)
12-17 08:54:39.941  5827  5827 E lfe     :      at android.os.Looper.loop(Looper.java:154)
12-17 08:54:39.941  5827  5827 E lfe     :      at android.os.HandlerThread.run(HandlerThread.java:61)
12-17 08:54:39.941  5827  5827 E lfe     : Caused by: java.lang.IllegalStateException: Playback stuck buffering and not loading
12-17 08:54:39.941  5827  5827 E lfe     :      at c.b.a.a.m0.l()
12-17 08:54:39.941  5827  5827 E lfe     :      ... 4 more
12-17 08:54:39.941  5827  5827 E lfe     : PlaybackFragment Player Error Unexpected Runtime Exception
12-17 08:54:39.941  5827  5827 E lfe     : Unexpected runtime error
12-17 08:54:39.941  5827  5827 E lfe     : Playback stuck buffering and not loading

*** skipped again, playback failed message, selected continue

12-17 08:55:15.677  5827  5827 I lfe     : PlaybackFragment Playing offset mSec:0
12-17 08:55:16.699  5827  5827 I lfe     : PlaybackFragment File Length changed from -1 to 2178983920
12-17 08:55:23.859  5827  5827 E lfe     : PlaybackFragment Player Error Question Time http://192.168.1.6:6544/Content/GetFile?StorageGroup=Default&FileName=/20101_20201210224300.ts
12-17 08:55:23.859  5827  5827 E lfe     : c.b.a.a.k0: Unexpected runtime error
12-17 08:55:23.859  5827  5827 E lfe     :      at c.b.a.a.m0.handleMessage()
12-17 08:55:23.859  5827  5827 E lfe     :      at android.os.Handler.dispatchMessage(Handler.java:98)
12-17 08:55:23.859  5827  5827 E lfe     :      at android.os.Looper.loop(Looper.java:154)
12-17 08:55:23.859  5827  5827 E lfe     :      at android.os.HandlerThread.run(HandlerThread.java:61)
12-17 08:55:23.859  5827  5827 E lfe     : Caused by: java.lang.IllegalStateException: Playback stuck buffering and not loading
12-17 08:55:23.859  5827  5827 E lfe     :      at c.b.a.a.m0.l()
12-17 08:55:23.859  5827  5827 E lfe     :      ... 4 more
12-17 08:55:23.860  5827  5827 E lfe     : PlaybackFragment Player Error Unexpected Runtime Exception
12-17 08:55:23.860  5827  5827 E lfe     : Unexpected runtime error
12-17 08:55:23.860  5827  5827 E lfe     : Playback stuck buffering and not loading
12-17 08:55:23.865  5827  5827 I lfe     : PlaybackFragment Playing offset mSec:0
12-17 08:55:24.878  5827  5827 I lfe     : PlaybackFragment File Length changed from -1 to 2178983920
12-17 08:55:32.035  5827  5827 E lfe     : PlaybackFragment Player Error Question Time http://192.168.1.6:6544/Content/GetFile?StorageGroup=Default&FileName=/20101_20201210224300.ts
12-17 08:55:32.035  5827  5827 E lfe     : c.b.a.a.k0: Unexpected runtime error
12-17 08:55:32.035  5827  5827 E lfe     :      at c.b.a.a.m0.handleMessage()
12-17 08:55:32.035  5827  5827 E lfe     :      at android.os.Handler.dispatchMessage(Handler.java:98)
12-17 08:55:32.035  5827  5827 E lfe     :      at android.os.Looper.loop(Looper.java:154)
12-17 08:55:32.035  5827  5827 E lfe     :      at android.os.HandlerThread.run(HandlerThread.java:61)
12-17 08:55:32.035  5827  5827 E lfe     : Caused by: java.lang.IllegalStateException: Playback stuck buffering and not loading
12-17 08:55:32.035  5827  5827 E lfe     :      at c.b.a.a.m0.l()
12-17 08:55:32.035  5827  5827 E lfe     :      ... 4 more
12-17 08:55:32.035  5827  5827 E lfe     : PlaybackFragment Player Error Unexpected Runtime Exception
12-17 08:55:32.035  5827  5827 E lfe     : Unexpected runtime error
12-17 08:55:32.035  5827  5827 E lfe     : Playback stuck buffering and not loading

*** playack failed again, selected continue

12-17 08:55:46.390  5827  5827 I lfe     : PlaybackFragment Playing offset mSec:0
12-17 08:55:47.468  5827  5827 I lfe     : PlaybackFragment File Length changed from -1 to 2178983920
12-17 08:55:54.434  5827  5827 E lfe     : PlaybackFragment Player Error Question Time http://192.168.1.6:6544/Content/GetFile?StorageGroup=Default&FileName=/20101_20201210224300.ts
12-17 08:55:54.434  5827  5827 E lfe     : c.b.a.a.k0: Unexpected runtime error
12-17 08:55:54.434  5827  5827 E lfe     :      at c.b.a.a.m0.handleMessage()
12-17 08:55:54.434  5827  5827 E lfe     :      at android.os.Handler.dispatchMessage(Handler.java:98)
12-17 08:55:54.434  5827  5827 E lfe     :      at android.os.Looper.loop(Looper.java:154)
12-17 08:55:54.434  5827  5827 E lfe     :      at android.os.HandlerThread.run(HandlerThread.java:61)
12-17 08:55:54.434  5827  5827 E lfe     : Caused by: java.lang.IllegalStateException: Playback stuck buffering and not loading
12-17 08:55:54.434  5827  5827 E lfe     :      at c.b.a.a.m0.l()
12-17 08:55:54.434  5827  5827 E lfe     :      ... 4 more
12-17 08:55:54.435  5827  5827 E lfe     : PlaybackFragment Player Error Unexpected Runtime Exception
12-17 08:55:54.435  5827  5827 E lfe     : Unexpected runtime error
12-17 08:55:54.435  5827  5827 E lfe     : Playback stuck buffering and not loading
12-17 08:55:54.440  5827  5827 I lfe     : PlaybackFragment Playing offset mSec:0
12-17 08:55:55.468  5827  5827 I lfe     : PlaybackFragment File Length changed from -1 to 2178983920
12-17 08:56:02.474  5827  5827 E lfe     : PlaybackFragment Player Error Question Time http://192.168.1.6:6544/Content/GetFile?StorageGroup=Default&FileName=/20101_20201210224300.ts
12-17 08:56:02.474  5827  5827 E lfe     : c.b.a.a.k0: Unexpected runtime error
12-17 08:56:02.474  5827  5827 E lfe     :      at c.b.a.a.m0.handleMessage()
12-17 08:56:02.474  5827  5827 E lfe     :      at android.os.Handler.dispatchMessage(Handler.java:98)
12-17 08:56:02.474  5827  5827 E lfe     :      at android.os.Looper.loop(Looper.java:154)
12-17 08:56:02.474  5827  5827 E lfe     :      at android.os.HandlerThread.run(HandlerThread.java:61)
12-17 08:56:02.474  5827  5827 E lfe     : Caused by: java.lang.IllegalStateException: Playback stuck buffering and not loading
12-17 08:56:02.474  5827  5827 E lfe     :      at c.b.a.a.m0.l()
12-17 08:56:02.474  5827  5827 E lfe     :      ... 4 more
12-17 08:56:02.474  5827  5827 E lfe     : PlaybackFragment Player Error Unexpected Runtime Exception
12-17 08:56:02.474  5827  5827 E lfe     : Unexpected runtime error
12-17 08:56:02.474  5827  5827 E lfe     : Playback stuck buffering and not loading

*** playback failed again, continue

12-17 08:56:13.752  5827  5827 I lfe     : PlaybackFragment Playing offset mSec:0
12-17 08:56:14.769  5827  5827 I lfe     : PlaybackFragment File Length changed from -1 to 2178983920
12-17 08:56:21.740  5827  5827 E lfe     : PlaybackFragment Player Error Question Time http://192.168.1.6:6544/Content/GetFile?StorageGroup=Default&FileName=/20101_20201210224300.ts
12-17 08:56:21.740  5827  5827 E lfe     : c.b.a.a.k0: Unexpected runtime error
12-17 08:56:21.740  5827  5827 E lfe     :      at c.b.a.a.m0.handleMessage()
12-17 08:56:21.740  5827  5827 E lfe     :      at android.os.Handler.dispatchMessage(Handler.java:98)
12-17 08:56:21.740  5827  5827 E lfe     :      at android.os.Looper.loop(Looper.java:154)
12-17 08:56:21.740  5827  5827 E lfe     :      at android.os.HandlerThread.run(HandlerThread.java:61)
12-17 08:56:21.740  5827  5827 E lfe     : Caused by: java.lang.IllegalStateException: Playback stuck buffering and not loading
12-17 08:56:21.740  5827  5827 E lfe     :      at c.b.a.a.m0.l()
12-17 08:56:21.740  5827  5827 E lfe     :      ... 4 more
12-17 08:56:21.740  5827  5827 E lfe     : PlaybackFragment Player Error Unexpected Runtime Exception
12-17 08:56:21.740  5827  5827 E lfe     : Unexpected runtime error
12-17 08:56:21.740  5827  5827 E lfe     : Playback stuck buffering and not loading
12-17 08:56:21.745  5827  5827 I lfe     : PlaybackFragment Playing offset mSec:0
12-17 08:56:22.761  5827  5827 I lfe     : PlaybackFragment File Length changed from -1 to 2178983920
12-17 08:56:29.635  5827  5827 E lfe     : PlaybackFragment Player Error Question Time http://192.168.1.6:6544/Content/GetFile?StorageGroup=Default&FileName=/20101_20201210224300.ts
12-17 08:56:29.635  5827  5827 E lfe     : c.b.a.a.k0: Unexpected runtime error
12-17 08:56:29.635  5827  5827 E lfe     :      at c.b.a.a.m0.handleMessage()
12-17 08:56:29.635  5827  5827 E lfe     :      at android.os.Handler.dispatchMessage(Handler.java:98)
12-17 08:56:29.635  5827  5827 E lfe     :      at android.os.Looper.loop(Looper.java:154)
12-17 08:56:29.635  5827  5827 E lfe     :      at android.os.HandlerThread.run(HandlerThread.java:61)
12-17 08:56:29.635  5827  5827 E lfe     : Caused by: java.lang.IllegalStateException: Playback stuck buffering and not loading
12-17 08:56:29.635  5827  5827 E lfe     :      at c.b.a.a.m0.l()
12-17 08:56:29.635  5827  5827 E lfe     :      ... 4 more
12-17 08:56:29.635  5827  5827 E lfe     : PlaybackFragment Player Error Unexpected Runtime Exception
12-17 08:56:29.635  5827  5827 E lfe     : Unexpected runtime error
12-17 08:56:29.635  5827  5827 E lfe     : Playback stuck buffering and not loading

I tried a few more times after this and kept getting playback failed.
This is again a HD BBC recording.  I also tried an SD recording and it plays perfectly without issue.

**Comment by bennettpeter** [comment]
The sample file you gave me earlier is 2 mins 30 seconds. I tried again and cannot reproduce the problem with that. There is an unexpected exception at the beginning, which is expected (!). After that I can skip back and forward as much as I like without any exception occurring. The response to skipping is rather slow, some 5 to 10 seconds before it resumes, but I see no exceptions apart from the one at the start. I set my skip forward and back to 10 seconds so that I could do a bunch of skips without hitting the end.

Please let me know how I can recreate this. I can attempt a fix but if I can't recreate it I can't test the fix. Can you tell me how to recreate it on that sample, or give me a sample where the problem occurs.

**Comment by bobMuk** [comment]
Hi Peter - appreciate you taking the time to look at this.

After some more playing around I started to think that the problem was occurring when the tail-end of a programme is at the start of the recording and I'm skipping from there to the start of the actual programme.  Which would explain why the clip that I uploaded previously didn't fail for you - I'd specified a skip of a few hundred MB with "dd" when getting the clip.
So I just took another clip - the first 4 minutes/150MB, so 2 minutes of the pre-programme and 2 minutes of the actual programme - but this time I copied it into myth's videos directory so I could play it through the app.  Skipping around the clip does not cause the issue.  As a sanity check I copied the whole recording into the directory too and skipping around the first 4 minutes *does* cause the issue.
So I'm at a loss - it'd be impractical to upload the whole recording somewhere.  I'll see if I can get a shorter recording that exhibits the issue, if I can't then I guess it's time to give up.

**Comment by bennettpeter** [comment]
Please try the new version leanfront-v0-245-gbc76542. I have made some changes to the way unexpected runtime errors are handled. I still expect to see one runtime error at the start of playback with these files that have a bad alternate audio track, but hopefully no more when skipping back and forward.

**Comment by bobMuk** [comment]
Unfortunately I'm still seeing the runtime exceptions when skipping back and forward, and there's still a long delay resuming playback once I've skipped (10-ish seconds).  Also (with 236 as well as 252 which I've just installed) sometimes when I skip, the playback seems to reset to the beginning, so the skip then starts from time index 0 (and so you then have to try and skip back to where you were in the first place).
Thanks again for spending time looking at this - but I think I'm going to have to concede that at least for now it's not going to work with UK Freeview HD, and investigate other ways of running a frontend.
